main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -17,21 +17,58 @@ import sbt.Def._
 import sbt.Keys._
 import sbt.ScopeFilter.Make._
 import sbt.SlashSyntax0._
+import sbt.StandardMain.exchange
 import sbt.internal.bsp._
 import sbt.internal.langserver.ErrorCodes
 import sbt.internal.protocol.JsonRpcRequestMessage
 import sbt.librarymanagement.Configuration
 import sbt.util.Logger
+import sjsonnew.shaded.scalajson.ast.unsafe.JNull
 import sjsonnew.shaded.scalajson.ast.unsafe.JValue
 import sjsonnew.support.scalajson.unsafe.Converter
 
 import scala.util.control.NonFatal
 
 object BuildServerProtocol {
   import sbt.internal.bsp.codec.JsonProtocol._
+
   private val capabilities = BuildServerCapabilities(
     CompileProvider(BuildServerConnection.languages),
-    dependencySourcesProvider = true
+    dependencySourcesProvider = true,
+    canReload = true
+  )
+
+  private val reload = "bspReload"
+  private val reloadFailed = "bspReloadFailed"
+  private val reloadSucceed = "bspReloadSucceed"
```

Comment:
I think it would be better to name this as `bspRealod` since there is a command named `reload`. It's a bit wordier but less ambiguous.